examples/hello-world/gnn-pt/jobs/gnn-pt/app/custom/pt_model_locator.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import List, Union
+import os.path as osp
+
+import torch.cuda
+from pt_constants import PTConstants
+from nvflare.apis.dxo import DXO
+from nvflare.apis.fl_context import FLContext
+from nvflare.app_common.abstract.model import model_learnable_to_dxo
+from nvflare.app_common.abstract.model_locator import ModelLocator
+from nvflare.app_opt.pt.model_persistence_format_manager import PTModelPersistenceFormatManager
+from torch_geometric.data import Batch
+from torch_geometric.datasets import PPI
+from torch_geometric.nn import GraphSAGE
+
+
+class PTModelLocator(ModelLocator):
+    def __init__(self):
+        super().__init__()
+
+        path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'PPI')
+        train_dataset = PPI(path, split='train')
+        train_data = Batch.from_data_list(train_dataset)
+        self.model = GraphSAGE(
+            in_channels=train_dataset.num_features,
```

Comment:
These lines are used to determine the "in_channels" of the "GraphSAGE" model.

However, you are assuming the dataset also exists on server side.

There should be a way to send this information to server or pre-configured it.
@holgerroth @ZiyueXu77 can you advise on how you do this, thanks